### DISCUSSION TITLE
Gradle build fails with github version release snapshot [12365]

### DISCUSSION THREAD

**Discussion by tparys** [eProsima/Fast-DDS-Gen#237] [Ideas]
This may be a WONTFIX issue, but figured I'd raise it.

Attempting to build fastddsgen via the source tarballs listed at [https://github.com/eProsima/Fast-DDS-Gen/releases](https://github.com/eProsima/Fast-DDS-Gen/releases), or via the code download option in the repository results in a broken source tree.

Looks like part of the gradle build process calls git, and the git metadata is stripped from these downloads.

Checking out the code tag via "git clone" results in a normal build.

**Comment by richiware** [comment]
Sorry for the inconvenience but currently this behaviour is normal and there is no quick fix to make it work when you download the source code from the Github's release section.

This repository depends on a external repository using Git submodule. The gradle build file of this project always try to do a `git submodule update --init`. When you download the `tar.gz` file from release section, this file doesn't contain the subdirectory `git` and that command always fails.

We have in mind to change this dependency in the future.
